source/views/menus/menu-bonapp.tsx

```diff
@@ -70,234 +59,219 @@ function findCafeMessage(cafeInfo: CafeInfoType, now: Moment): string | null {
+
+	let ignoreMenus = dayparts[0]?.length ? ignoreProvidedMenus : true
+
+	return mealInfoItems.map((mealInfo) =>
+		prepareSingleMenu(mealInfo, foodItems, ignoreMenus),
+	)
+}
+
+function prepareFood(cafeMenu: MenuInfoType) {
+	return mapValues(cafeMenu.items, (item) => ({
+		...item, // we want to edit the item, not replace it
+		station: decode(toLaxTitleCase(trimStationName(item.station))), // <b>@station names</b> are a mess
+		label: decode(trimItemLabel(item.label)), // clean up the titles
+		description: innerTextWithSpaces(parseHtml(item.description || '')), // clean up the descriptions
+	}))
+}
+
+export function BonAppHostedMenu(props: Props): JSX.Element {
+	let [errorMessage, setErrorMessage] = useState<string | null>(null)
+	let [loading, setLoading] = useState<boolean>(true)
+	let [refreshing, setRefreshing] = useState<boolean>(false)
+	let [now, setNow] = useState<Moment>(moment.tz(timezone()))
+	let [cafeInfo, setCafeInfo] = useState<CafeInfoType | null>(null)
+	let [cafeMenu, setCafeMenu] = useState<MenuInfoType | null>(null)
+
+	useEffect(() => {
+		;(async () => {
+			setLoading(true)
+			await fetchData(props.cafe)
+			setLoading(false)
+		})()
+	}, [props.cafe])
+
+	let retry = () => {
+		;(async () => {
+			setLoading(true)
+			await fetchData(props.cafe)
+			setLoading(false)
+			setErrorMessage(null)
+		})()
 	}
 
-	fetchData = async (props: Props, reload?: boolean) => {
+	let fetchData = async (cafeThing: Props['cafe'], reload?: boolean) => {
 		let menuUrl
 		let cafeUrl
-		let cafe = typeof props.cafe === 'string' ? props.cafe : props.cafe.id
-		if (typeof props.cafe === 'string') {
+		let cafe = typeof cafeThing === 'string' ? cafeThing : cafeThing.id
+		if (typeof cafeThing === 'string') {
 			menuUrl = API(`/food/named/menu/${cafe}`)
 			cafeUrl = API(`/food/named/cafe/${cafe}`)
-		} else if ('id' in props.cafe) {
+		} else if ('id' in cafeThing) {
 			menuUrl = API(`/food/menu/${cafe}`)
 			cafeUrl = API(`/food/cafe/${cafe}`)
 		} else {
 			throw new Error('invalid cafe passed to BonappMenu!')
 		}
 
 		try {
-			let cafeMenuPromise: Promise<MenuInfoType> = fetch(menuUrl, {
-				delay: reload ? 500 : 0,
-			}).json()
+			let delay = reload ? 500 : 0
 
-			let cafeInfoPromise: Promise<CafeInfoType> = fetch(cafeUrl, {
-				delay: reload ? 500 : 0,
-			}).json()
+			let cafeMenuPromise = fetch(menuUrl, {delay}).json<MenuInfoType>()
+			let cafeInfoPromise = fetch(cafeUrl, {delay}).json<CafeInfoType>()
 
 			let [cafeMenu, cafeInfo] = await Promise.all([
 				cafeMenuPromise,
 				cafeInfoPromise,
 			])
 
-			this.setState(() => ({
-				cafeMenu: cafeMenu ? cafeMenu : null,
-				cafeInfo: cafeInfo ? cafeInfo : null,
-				now: moment.tz(timezone()),
-			}))
+			setCafeMenu(cafeMenu ?? null)
+			setCafeInfo(cafeInfo ?? null)
+			setNow(moment.tz(timezone()))
 		} catch (error) {
+			if (!(error instanceof Error)) {
+				setErrorMessage('Unknown Error: Not an Error')
+				return
+			}
```

Comment:
This is an interesting error case?

Reply:
Typescript treats a catch statement as `catch (err: unknown|any)`, so it complains that it doesn't know about the "message" property… so you have to check for instanceof Error first. 

Which technically makes sense - you can `throw` anything - but isn't a case I'd considered before.